cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaClusterTest.java

```diff
@@ -313,54 +316,56 @@ private void checkStatefulSet(StatefulSet ss, Kafka cm, boolean isOpenShift) {
         }
     }
 
+    // TODO test volume claim templates
+
     @Test
-    public void testDeleteClaim() {
-        Kafka assembly = ResourceUtils.createKafkaCluster(namespace, cluster, replicas,
-                image, healthDelay, healthTimeout, metricsCm, configuration, emptyMap());
-        KafkaCluster kc = KafkaCluster.fromCrd(assembly, VERSIONS);
-        StatefulSet ss = kc.generateStatefulSet(true);
-        assertFalse(KafkaCluster.deleteClaim(ss));
+    public void testPodNames() {
 
-        assembly = new KafkaBuilder(ResourceUtils.createKafkaCluster(namespace, cluster, replicas,
+        for (int i = 0; i < replicas; i++) {
+            assertEquals(KafkaCluster.kafkaClusterName(cluster) + "-" + i, kc.getPodName(i));
+        }
+    }
+
+    @Test
+    public void testPvcNames() {
+        Kafka assembly = new KafkaBuilder(ResourceUtils.createKafkaCluster(namespace, cluster, replicas,
                 image, healthDelay, healthTimeout, metricsCm, configuration, emptyMap()))
                 .editSpec()
                     .editKafka()
                         .withStorage(new PersistentClaimStorageBuilder().withDeleteClaim(false).withSize("100Gi").build())
                     .endKafka()
                 .endSpec()
                 .build();
-        kc = KafkaCluster.fromCrd(assembly, VERSIONS);
-        ss = kc.generateStatefulSet(true);
-        assertFalse(KafkaCluster.deleteClaim(ss));
+        KafkaCluster kc = KafkaCluster.fromCrd(assembly, VERSIONS);
+
+        List<PersistentVolumeClaim> pvcs = kc.getVolumeClaims();
+
+        for (int i = 0; i < replicas; i++) {
+            assertEquals(kc.VOLUME_NAME + "-" + KafkaCluster.kafkaPodName(cluster, i),
+                    pvcs.get(0).getMetadata().getName() + "-" + KafkaCluster.kafkaPodName(cluster, i));
```

Comment:
`assertEquals(kc.VOLUME_NAME, pvcs.get(0).getMetadata().getName())` would be clearer, no?

Reply:
Well, technically it's true because of the latest part `KafkaCluster.kafkaPodName(cluster, i)` which is the same but semantically we are comparing PVC names which includes the pod name (using the VCT name as "prefix")